src/formik.tsx

```diff
@@ -550,6 +560,9 @@ export class Formik<
   };
 
   handleReset = () => {
+    if (this.props.onReset) {
+      this.props.onReset(this.state.values, this.getFormikActions());
```

Comment:
Should we allow this to async? 

Some ideas:

- sniff for a promise
- expose a done() argument 
- force this to be sync

i think sniffing for a promise is the move. 

naive implementation:
```js
if (this.props.onReset) {
  const maybePromise = this.props.onReset(this.state.values, this.getFormikActions());
  if (isPromise(maybePromise)) {
    // do we pass values returned by the promise?? hmmm. 
    maybePromise.then(this.resetForm) 
  } else {
    this.resetForm();
 }
}
```

Reply:
```js
if (this.props.onReset) {
      const maybePromisedOnReset = (this.props.onReset as any)(
        this.state.values,
        this.getFormikActions()
      );
 
      if (isPromise(maybePromisedOnReset)) {
        (maybePromisedOnReset as Promise<any>).then(this.resetForm);
      } else {
        this.resetForm();
      }
    }
```
Maybe `onReset` returning new `nextValues` for `resetForm` will be useful ?